tests/models/demultiplexing/test_run_parameters.py

```diff
@@ -0,0 +1,180 @@
+    assert run_parameters.flow_cell_type == "novaseq"
+
+
+def test_get_unknown_type_run_parameters(unknown_run_parameters: Path, caplog):
+    """Test that reading a run_parameters with unknown flowcell file raises an error."""
+    caplog.set_level(logging.INFO)
+    # GIVEN a RunParameters object created from a run parameters file with unknown flowcell
+    run_parameters = RunParameters(unknown_run_parameters)
+
+    # WHEN fetching the flowcell type
+    with pytest.raises(FlowCellError):
+        # THEN assert that an exception was raised since the flowcell type was unknown
+        run_parameters.flow_cell_type
+
+    assert "Unknown flow cell type" in caplog.text
+
+
+def test_get_flow_cell_type_when_missing(run_parameters_missing_flowcell_type: Path, caplog):
+    """Test that reading a run_parameters with missing flowcell file raises an error."""
+    caplog.set_level(logging.INFO)
+    # GIVEN a RunParameters object created from a run parameters file with missing flowcell
+    run_parameters = RunParameters(run_parameters_missing_flowcell_type)
+
+    # WHEN parsing the run parameters
+    with pytest.raises(FlowCellError):
+        # THEN assert that an exception was raised since the flow cell type was unknown
+        run_parameters.flow_cell_type
+
+    assert "Could not determine flow cell type" in caplog.text
+
+
+def test_get_base_mask(novaseq_run_parameters: Path):
+    """Test that generating the run parameters base mask returns the expected result."""
+    # GIVEN a RunParameters object created from a valid run parameters file
+    run_parameters = RunParameters(novaseq_run_parameters)
+
+    # WHEN fetching the base mask
+    base_mask: str = run_parameters.get_base_mask()
+
+    # THEN the base mask has the correct structure
+    assert base_mask.startswith("Y")
+    assert ",I" in base_mask
+
+
+def test_get_base_mask_undetermined_cycles(run_parameters_missing_flowcell_type: Path, caplog):
+    """Test that generating the base mask from a file without cycle info fails."""
+    caplog.set_level(logging.INFO)
+    # GIVEN a RunParameters object created from a run parameters file with missing cycle info
+    run_parameters = RunParameters(run_parameters_missing_flowcell_type)
+
+    # WHEN creating the base mask
+    with pytest.raises(FlowCellError):
+        # THEN assert that an exception was raised since the cycle data was not found
+        run_parameters.get_base_mask()
+
+    assert "Could not determine length of reads one" in caplog.text
+
+
+def test_flow_cell_mode(novaseq_run_parameters: Path):
+    """Test that the flow cell mode is of a valid file in the list of expected values."""
+    # GIVEN a RunParameters object created from a valid run parameters file
+    run_parameters = RunParameters(novaseq_run_parameters)
+
+    # WHEN fetching the flow cell mode
+
+    # THEN the flow cell mode is in the list of expected values
+    assert run_parameters.flow_cell_mode in ["S2", "S4"]
```

Comment:
Hmm, maybe we should call this constant FlowCellMode and not SheetType.

Reply:
Yes I agree 👍

Reply:
Lets fix this constant use in your PR